File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/factory_method/SubFactory.java
package jv.com.tencent.cain.design_pattern.factory_method;

/**
 * @author cainjiang
 * @date 2019-05-20
 */
public class SubFactory implements IOperationFactory {
    @Override
    public Operation createOpration() {
        SubOperation subOperation = new SubOperation();
        subOperation.operationSymbol = "-";
        subOperation.operationName = "减法";
        return subOperation;
    }

}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/factory_method/SubFactory.kt
package kt.com.tencent.cain.desigin_pattern.factory_method

import kt.com.tencent.cain.desigin_pattern.simple_factoty.Operation
import kt.com.tencent.cain.desigin_pattern.simple_factoty.SubOperation

/**
 * @author cainjiang
 * @date 2019-05-21
 */
class SubFactory : IOperationFactory {
    override fun createOpration(): Operation {
        val subOperation = SubOperation()
        subOperation.operationSymbol = "-"
        subOperation.operationName = "减法"
        return subOperation
    }
}